cirq/contrib/quirk/url_to_circuit.py

```diff
@@ -77,16 +142,31 @@ def quirk_url_to_circuit(quirk_url: str) -> 'cirq.Circuit':
                 cell.modify_column(col)
 
     # Extract circuit operations from modified cells.
-    result = cirq.Circuit()
+    result = circuits.Circuit()
     for col in parsed_cols:
-        basis_change = cirq.Circuit(
+        basis_change = circuits.Circuit(
             cell.basis_change() for cell in col if cell is not None)
-        body = cirq.Circuit(
+        body = circuits.Circuit(
             cell.operations() for cell in col if cell is not None)
         result += basis_change
         result += body
         result += basis_change**-1
 
+    # Remap qubits if requested.
+    if qubits is not None:
+        qs = cast(Sequence['cirq.Qid'], qubits)
```

Comment:
IIRC, mypy is able to make the inference within a conditional with `is not None`. Is this not the case?

Reply:
The issue is that qubits is used inside the lambda, which involves capturing and deferred execution semantics. Mypy does not make the inference.